packages/lib/pcd-types/src/pcd.ts

```diff
@@ -369,10 +392,17 @@ export type RawValueType<T extends Argument<any, unknown>> =
     ? U
     : T;
 
-export type ArgumentValidator<T extends Argument<any, unknown>> = (
-  value: RawValueType<T>,
-  params: T["validatorParams"]
-) => boolean;
+/**
+ * Argument validator as a predicate taking both the argument's value and its
+ * validator parameters as inputs. In the case of a record argument, this is a
+ * mapping from record keys to such predicates for the record value type.
+ */
+export type ArgumentValidator<T extends Argument<any, unknown>> =
+  T extends RecordArgument<infer S, infer U>
+    ? (
+        s: S
+      ) => (value: RawValueType<U>, params: U["validatorParams"]) => boolean
```

Comment:
Could `s` (name) become an extra parameter here before value, rather than needing the extra layer of function call?